src/pudl/helpers.py

```diff
@@ -867,39 +867,43 @@ def month_year_to_date(df):
     return df
 
 
-def fix_leading_zero_gen_ids(df):
-    """Remove leading zeros from EIA generator IDs which are numeric strings.
+def remove_leading_zeros_from_numeric_strings(
+    df: pd.DataFrame, col_name: str
+) -> pd.DataFrame:
+    """Remove leading zeros frame column values that are numeric strings.
 
-    If the DataFrame contains a column named ``generator_id`` then that column
-    will be cast to a string, and any all numeric value with leading zeroes
-    will have the leading zeroes removed. This is necessary because in some
-    but not all years of data, some of the generator IDs are treated as integers
-    in the Excel spreadsheets published by EIA, so the same generator may show
-    up with the ID "0001" and "1" in different years.
+    Sometimes an ID column (like generator_id or unit_id) will be reported with leading
+    zeros and sometimes it won't. For example, in the Excel spreadsheets published by
+    EIA, the same generator may show up with the ID "0001" and "1" in different years
+    This function strips the leading zeros from those numeric strings so the data can
+    be mapped accross years and datasets more reliably.
 
     Alphanumeric generator IDs with leadings zeroes are not affected, as we
-    found no instances in which an alphanumeric generator ID appeared both with
-    and without leading zeroes.
+    found no instances in which an alphanumeric ID appeared both with
+    and without leading zeroes. The ID "0A1" will stay "0A1".
 
     Args:
-        df (pandas.DataFrame): DataFrame, presumably containing a column named
-            generator_id (otherwise no action will be taken.)
+        df: A DataFrame containing the column you'd like to remove numeric leading zeros
+            from.
+        col_name: The name of the column you'd like to remove numeric leading zeros
+            from.
 
     Returns:
-        pandas.DataFrame
+        A DataFrame without leading zeros for numeric string values in the desired
+        column.
 
     """
-    if "generator_id" in df.columns:
-        fixed_generator_id = (
-            df["generator_id"]
-            .astype(str)
-            .apply(lambda x: re.sub(r"^0+(\d+$)", r"\1", x))
-        )
-        num_fixes = len(df.loc[df["generator_id"].astype(str) != fixed_generator_id])
-        logger.debug("Fixed %s EIA generator IDs with leading zeros.", num_fixes)
-        df = df.drop("generator_id", axis="columns").assign(
-            generator_id=fixed_generator_id
-        )
+    if col_name in df.columns:
+        number_with_leading_zeros = r"^0+\d+$"
+        if df[col_name].str.contains(number_with_leading_zeros).any():
+            logger.debug(f"Fixing leading zeros in {col_name} column")
+            df.loc[
+                df[col_name].notna()
+                & df[col_name].str.contains(number_with_leading_zeros),
```

Comment:
You're performing this probably costly operation twice, instead of storing the value and reusing it. Instead I think you could do:
```py
leading_zeros = df[col_name].str.contains(number_with_leading_zeros).fillna(False)
if leading_zeros.any():
        logger.debug(f"Fixing leading zeros in {col_name} column")
        df.loc[leading_zeros, col_name] = df[col_name].str.replace(r"^0+", "", regex=True)
```